ResearchKit/Common/ORKAxisView.m

```diff
@@ -0,0 +1,148 @@
+ specific prior written permission. No license is granted to the trademarks of 
+ the copyright holders even if such marks are included in this software. 
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+ 
+#import "ORKAxisView.h"
+
+
+@interface ORKAxisView ()
+
+@property (nonatomic, strong) NSMutableArray *titleLabels;
+
+@end
+
+
+@implementation ORKAxisView {
+    
+    NSArray *_constraints;
+    CGFloat _lastLabelPadding;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (void)sharedInit {
+    _titleLabels = [NSMutableArray new];
+    _lastLabelPadding = 10;
+}
+
+- (void)updateConstraints {
+    [super updateConstraints];
+    [NSLayoutConstraint deactivateConstraints:_constraints];
+    _constraints = nil;
+    
+    NSMutableArray *constraints = [@[]mutableCopy];
+    CGFloat segmentWidth = CGRectGetWidth(self.bounds) / (self.titleLabels.count - 1);
+    
+    for (NSUInteger i = 0; i < self.titleLabels.count; i++) {
+        UILabel *label = self.titleLabels[i];
+        CGFloat offset = i * segmentWidth;
+        [constraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                            attribute:NSLayoutAttributeCenterY
+                                                            relatedBy:NSLayoutRelationEqual
+                                                               toItem:label.superview
+                                                            attribute:NSLayoutAttributeCenterY
+                                                           multiplier:1
+                                                             constant:0]];
+        [constraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                            attribute:NSLayoutAttributeCenterX
+                                                            relatedBy:NSLayoutRelationEqual
+                                                               toItem: label.superview
```

Comment:
Avoid space here before `label`, and before `-` in the two following constraints.